cmd/osm-controller/osm-controller.go

```diff
@@ -103,6 +103,7 @@ func init() {
 
 	// feature flags
 	flags.BoolVar(&optionalFeatures.WASMStats, "stats-wasm-experimental", false, "Enable a WebAssembly module that generates additional Envoy statistics.")
+	flags.BoolVar(&optionalFeatures.VerboseLogging, "verbose-logging", false, "Enable VerboseLogging to log more metadata on Pods (UID vs Name), Certificates (SerialNumber vs CommonName) etc.")
```

Comment:
configmap value maybe?

Reply:
For this particular flag we want to be a bit stricter. Because when you enable this we would be logging so much more context around each Pod and Certificate etc. we should be careful not to provide a way for this to be ambiguously enabled in production and start logging things we should not.

Reply:
That's a good point. I suggest we rework the log messages and ensure what we log in production environments is sufficient to debug basic issues. I see value in this flag enabling more verbose logging in dev environments, but we should really be thinking about how to extract the most in production because that's where the challenge is. I don't believe there is anything wrong with logging k8s resource names for non-sensitive resources (not secrets, certs etc.). 

Since this is behind a feature flag, it seems reasonable to experiment this approach.